src/operator/contrib/sync_batch_norm-inl.h

```diff
@@ -275,14 +275,17 @@ class SyncBatchNorm : public Operator {
       static_cast<real_t>(in_data[syncbatchnorm::kData].shape_.Size());
     Tensor<xpu, 4> data;
     Tensor<xpu, 4> out;
-    if (in_data[syncbatchnorm::kData].ndim() == 2) {
+    if (in_data[syncbatchnorm::kData].ndim() == 4) {
```

Comment:
can you explain why you change ndim special case from 2 to 4?

Reply:
Yes. In 3DConvNet, the input dimension of BatchNorm is 5 (NCDHW). However, SyncBatchNorm only supported 2D or 4D inputs before, and BatchNorm supports all dimensions. So I modified the code of SyncBatchNorm to support all dimensions of inputs.

I inverse the branches of if-else ndim==4. When ndim is 4, we get the TBlob directly, otherwise we reshape the TBlob to dshape (N,C,1,D*H*W).